### ISSUE TITLE
Overhaul this package in order to be used with other robots

### ISSUE THREAD

**Issue by fmessmer** [ipa320/cob_calibration#12]
This package was never really usable with any robot other than cob3-3 - despite all efforts!
It needs a complete overhaul possibly using the upstream `calibration` (official) or "industrial calibration" packages rather than the copy-pasted cob-specific packages!

@fmessmer FYI

**Comment by ipa-rmb** [comment]
My student Marc implemented a totally new approach that should be easier to transfer. We tested it with a Festo Robotino and raw3-1 so far. It requires the laser scanner(s) to be well-calibrated in their relative position to base_link and then can automatically tell you camera mounting positions wrt. to base_link and calibrates the arm against a camera.

I would suggest to completely delete the current software from the indigo_dev branch (we keep it at the older branches) and replace it with our new. What do you think?

**Comment by fmessmer** [comment]
that sounds interesting...
where can the code be found? is there a pr for discussion? is there anything useful left from the current cob_calibration?
can you/your student provide a documentation/readme/demonstration?

**Comment by ipa-rmb** [comment]
The student just succeeded this week with a test calibration on raw3-1 with the old arm/torso setup. He will focus on code cleanup, simplification and generalization within the next weeks and update our documentation at http://wiki.ros.org/robotino_calibration . A PR will be opened once he has finished the cleanup and documentation work (expect this will be in ca. two months from now).

I don't know whether there is anything useful left from the current code. Unfortunately, documentation is not so telling. Maybe cob_laser_calibration could be useful, we do not calibrate laser scanners but expect they are already well-calibrated instead. Anyways, all of the code is written in Python, which is hard for me to overlook. I would just go with our new code as said above.

If laser calibration is a topic of interest, I could ask my student Marc whether he would like to approach this afterwards or within a thesis.

**Comment by fmessmer** [comment]
a test on cob4 - proofing the portability would be cool :wink:

have you investigated http://wiki.ros.org/industrial_extrinsic_cal

**Comment by ipa-rmb** [comment]
Yes sure, a test on cob 4 is the next job of Marc after cleaning up the code.

Yes again, we checked http://wiki.ros.org/industrial_extrinsic_cal but the tutorials looked like the calibration package is rather concerned with static cameras not placed on a mobile robot. I might be wrong with that assessment, but it looked like to much effort to us to make that package usable for our purposes. Probably, it is doing a few things quite similar to our package. Having our own package, however, allows us more control to fully adapt things to our needs.

As an example: we can calibrate the moving robot against a static environment in which we place multiple markers on different surfaces. I am not totally sure whether the package supports multiple calibration targets at the same time. Second, we have to optimize several unknown transformations in the tf chain at the same time, e.g. mounting pose of a camera on a pan-tilt unit, whose mounting pose relative to base_link is also unknown. I think we will support multiple of these unknown transforms.

I can ask Marc to check whether the referenced package could do the same, but since we are almost finished anyways, I am also ok with just taking our development which might be less powerful than http://wiki.ros.org/industrial_extrinsic_cal but simpler to use with (and tailored towards) a typical mobile manipulator robot.

**Comment by ipa-rmb** [comment]
FYI @ipa-rmb-mr . You may add more information on our package if you feel like.

**Comment by mr90x** [comment]
I am on holidays until 22st August to prepare for exams. Afterwards, I will focus on cleanup and documentation.
